sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -702,13 +702,14 @@ private static RowSignature analysisToRowSignature(final SegmentAnalysis analysi
 
   Map<SegmentId, AvailableSegmentMetadata> getSegmentMetadataSnapshot()
   {
-    final Map<SegmentId, AvailableSegmentMetadata> segmentMetadata = new HashMap<>();
     synchronized (lock) {
+      final Map<SegmentId, AvailableSegmentMetadata> segmentMetadata = Maps.newHashMapWithExpectedSize(
+          segmentMetadataInfo.values().size());
```

Comment:
should this be `segmentMetadataInfo.values().stream().mapToInt(v -> v.size()).sum()` ?